models/git/branch_list.go

```diff
@@ -63,30 +64,83 @@ func (branches BranchList) LoadPusher(ctx context.Context) error {
 	return nil
 }
 
+func (branches BranchList) LoadRepo(ctx context.Context) error {
+	ids := container.Set[int64]{}
+	for _, branch := range branches {
+		if branch.RepoID > 0 {
```

Comment:
```suggestion
		if branch.RepoID > 0 && branch.Repo == nil {
```

Reply:
we should also add this check in the below, or we will get bug here.
Done.